Helper/Discount.php

```diff
@@ -1275,6 +1283,39 @@ public function loadMagentoGiftCardAccount($code, $websiteId)
         return $result;
     }
     
+    /**
+     * Get the Magento_GiftCardAccount Gift Card data from quote
+     *
+     * @param Quote $quote
+     *
+     * @return array
+     */
+    public function getMagentoGiftCardAccountGiftCardData($quote)
+    {
+        if (! $this->isMagentoGiftCardAccountAvailable()) {
+            return [];
+        }
+        /** @var \Magento\GiftCardAccount\Helper\Data */
+        $giftCardAccountHelper = $this->moduleGiftCardAccountHelper->getInstance();
+        
+        if (!$giftCardAccountHelper) {
+            return [];
+        }
+        
+        $cards = $giftCardAccountHelper->getCards($quote);
+
+        if (!$cards) {
+            $cards = [];
+        } else {
+            $cards = array_column($cards,
+                                  'a', // \Magento\GiftCardAccount\Model\Giftcardaccount::AMOUNT,
+                                  'c'  // \Magento\GiftCardAccount\Model\Giftcardaccount::CODE
```

Comment:
Why not using the constants instead of adding them as a comment?

Reply:
Cause we experience some issues if the other plugin abandon the related constants or move them to another class, I think it is unreliable to use constants from other plugins.

Reply:
@danac-gs I have updated the code to cover cases both